js/latoken.js

```diff
@@ -0,0 +1,715 @@
+            let frozen = this.safeValue(balance, 'frozen'); 
+            let pending = this.safeValue(balance, 'pending');
+            result.push({
+                'currencyId': currencyId,
+                'symbol': symbol,
+                'name': name,
+                'amount': amount,
+                'available': available,
+                'frozen': frozen,
+                'pending': pending,
+            });
+        }
+        return result;
+    }
+
+    parseBidAsk (bidask) {
+        let price = parseFloat (bidask['price'])
+        let amount = parseFloat (bidask['amount'])
+        return { 
+            'price': price,
+            'amount': amount, 
+        }
+    }
+
+    parseBidsAsks (bidasks, priceKey = 0, amountKey = 1) {
+        return Object.values (bidasks || []).map (bidask => this.parseBidAsk (bidask, priceKey, amountKey))
+    }
+
+    parseOrderBook (orderbook, bidsKey = 'bids', asksKey = 'asks', priceKey = 0, amountKey = 0) {
+        return {
+            'pairId': orderbook['pairId'],
+            'symbol': orderbook['symbol'],
+            'spread': orderbook['spread'],
+            'asks': (asksKey in orderbook) ? this.parseBidsAsks (orderbook[asksKey], priceKey, amountKey) : null,
+            'bids': (bidsKey in orderbook) ? this.parseBidsAsks (orderbook[bidsKey], priceKey, amountKey) : null,
+        }
+    }
+
+    async fetchOrderBook (symbol, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market(symbol);
+        let request = {
+            'symbol': market['symbol'],
+        };
+        let response = await this.publicGetMarketDataOrderBook (request, params);
+        let orderbook = this.parseOrderBook(response);
+        return orderbook;
+    }
+
+    parseTicker( ticker, market = undefined) {
+        let symbol = this.findSymbol(this.safeString(ticker, 'symbol'), market);
+        return {
+            'id': this.safeValue(ticker, 'pairId'),
+            'symbol': symbol,
+            'volume': this.safeFloat(ticker, 'volume'),
+            'openPrice': this.safeFloat(ticker, 'open'),
+            'lowPrice': this.safeFloat(ticker, 'low'),
+            'highPrice': this.safeFloat(ticker, 'high'),
+            'closePrice': this.safeFloat(ticker, 'close'),
+            'priceChange': this.safeFloat(ticker, 'priceChange'),
+        }
+    }
+
+    async fetchTicker(symbol, params = {}) {
+        await this.loadMarkets();
+        let market = this.market(symbol);
+        let response = await this.publicGetMarketDataTicker( this.extend({
+            'symbol': market['symbol'], 
+        }, params));
+        return this.parseTicker(response, market);
+    }
+
+    async fetchCurrencies(params = undefined){
+        if (params !== undefined) {
+            const currencies = await this.publicGetExchangeInfoCurrencies(params);
+            let result = {};      
+            result.id = currencies['currencyId'];
```

Comment:
regular variables should be used here instead of making a result object

Reply:
done